homeassistant/components/octoprint/__init__.py

```diff
@@ -56,14 +68,20 @@ def ensure_valid_path(value):
 
 SENSOR_TYPES = {
     # API Endpoint, Group, Key, unit, icon
-    'Temperatures': ['printer', 'temperature', '*', TEMP_CELSIUS],
-    "Current State": ['printer', 'state', 'text', None, 'mdi:printer-3d'],
-    "Job Percentage": ['job', 'progress', 'completion', '%',
-                       'mdi:file-percent'],
-    "Time Remaining": ['job', 'progress', 'printTimeLeft', 'seconds',
-                       'mdi:clock-end'],
-    "Time Elapsed": ['job', 'progress', 'printTime', 'seconds',
-                     'mdi:clock-start'],
+    'Temperatures': ['printer', ['temperature'],
+                     TEMP_CELSIUS, 'mdi:thermometer'],
+    "Current State": ['printer', ['state', 'text'],
+                      None, 'mdi:printer-3d'],
+    "Job Percentage": ['job', ['progress', 'completion'],
+                       '%', 'mdi:file-percent'],
+    "Time Remaining": ['job', ['progress', 'printTimeLeft'],
+                       'seconds', 'mdi:clock-end'],
```

Comment:
We should not put relative time in the state machine. Instead, make it a timestamp and add the timestamp device class.

Reply:
(that way, Lovelace can render it as a countdown, timestamp etc)